packages/agreement-readmodel-writer/test/agreement-readmodel-writer.V1.integration.test.ts

```diff
@@ -0,0 +1,290 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { generateMock } from "@anatine/zod-mock";
+import {
+  AgreementCollection,
+  ReadModelRepository,
+  readModelWriterConfig,
+} from "pagopa-interop-commons";
+import {
+  getMockAgreement,
+  mongoDBContainer,
+  writeInReadmodel,
+} from "pagopa-interop-commons-test";
+import {
+  AgreementAddedV1,
+  AgreementConsumerDocumentAddedV1,
+  AgreementConsumerDocumentRemovedV1,
+  AgreementContractAddedV1,
+  AgreementDeletedV1,
+  AgreementDocument,
+  AgreementEventEnvelope,
+  AgreementStateV1,
+  AgreementUpdatedV1,
+  generateId,
+  toReadModelAgreement,
+} from "pagopa-interop-models";
+import { StartedTestContainer } from "testcontainers";
+import { afterAll, afterEach, beforeAll, describe, expect, it } from "vitest";
+import { handleMessageV1 } from "../src/consumerServiceV1.js";
+import { toAgreementDocumentV1 } from "./protobufConverterToV1.js";
+
+describe("events V1", async () => {
+  let agreements: AgreementCollection;
+  let startedMongoDBContainer: StartedTestContainer;
+
+  const config = readModelWriterConfig();
+  beforeAll(async () => {
+    startedMongoDBContainer = await mongoDBContainer(config).start();
+
+    config.readModelDbPort = startedMongoDBContainer.getMappedPort(27017);
+
+    const readModelRepository = ReadModelRepository.init(config);
+    agreements = readModelRepository.agreements;
+  });
+
+  afterEach(async () => {
+    await agreements.deleteMany({});
+  });
+
+  afterAll(async () => {
+    await startedMongoDBContainer.stop();
+  });
+
```

Comment:
```suggestion
```
You can delete all this initial setup in boths files by adding the following `utils.ts` and relative configs:

    import { setupTestContainersVitest } from "pagopa-interop-commons-test";
    import { afterEach, inject } from "vitest";
    
    export const { cleanup, readModelRepository } = setupTestContainersVitest(
      inject("readModelConfig")
    );
    
    afterEach(cleanup);
    
    export const agreements = readModelRepository.agreements;